projects/count-up/src/lib/count-up.directive.ts

```diff
@@ -45,18 +45,19 @@ export class CountUpDirective implements OnChanges {
     if (!isPlatformBrowser(this.platformId)) {
       return;
     }
-
+  
     const { options, endVal } = changes;
-    if (endVal?.currentValue !== undefined) {
-      if (this.countUp !== undefined) {
-        this.zone.runOutsideAngular(() => {
+  
+    if (this.countUp !== undefined) {
+      this.zone.runOutsideAngular(() => {
+        if (endVal?.currentValue !== undefined) {
           this.countUp.update(this.endVal);
-        });
-      } else {
-        this.initAndRun();
-      }
-    }
-    else if (options?.currentValue !== undefined) {
+        }
+        if (options?.currentValue !== undefined) {
```

Comment:
Make this `else if` since it completely reinitializes countUp and makes the update call moot.

Reply:
Here's an enhancement that I think would be nice. We can check if the options specify a start value, and if not, use the current value of countUp so that it effectively does an update. Just below the else if add:
```js
if (!this.options.startVal) {
    this.options.startVal = this.countUp.frameVal;
}
```

Reply:
I also realized that `runOutsideAngular` should only wrap the update call

Reply:
Had some issues implementing these changes as suggested, was seeing the original issue pop up again. I reworked the logic a bit to capture your suggestions and simplify things a bit, seems to be working now. Let me know what you think!